src/wallet/wallet.cpp

```diff
@@ -1684,7 +1697,15 @@ CBlockIndex* CWallet::ScanForWalletTransactions(CBlockIndex* pindexStart, CBlock
             if (pindex == pindexStop) {
                 break;
             }
-            pindex = chainActive.Next(pindex);
+            {
+                LOCK(cs_main);
+                pindex = chainActive.Next(pindex);
+                if (pindex && !chainActive.Contains(pindex)) break;
+                if (tip != chainActive.Tip()) {
+                    // in case the tip has changed, update progress max
+                    dProgressTip = GuessVerificationProgress(chainParams.TxData(), tip);
```

Comment:
err, didnt you mean to set tip to chainActive.Tip() first?

Reply:
Argh. Thanks. Fixed.